src/figuresClasses.ts

```diff
@@ -1,11 +1,89 @@
-export interface Figure {}
+type ShapeType = 'Triangle' | 'Circle' | 'Rectangle';
+type ColorType = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: ShapeType;
+  color: ColorType;
 
-export class Circle implements Figure {}
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: ShapeType = 'Triangle';
+
+  constructor(
+    public color: ColorType,
+    public sideA: number,
+    public sideB: number,
+    public sideC: number,
+  ) {
+    if (sideA <= 0 || sideB <= 0 || sideC <= 0) {
+      throw new Error('All sides must be greater than zero');
+    }
+
+    if (
+      Math.max(sideA, sideB, sideC) >=
+      sideA + sideB + sideC - Math.max(sideA, sideB, sideC)
+    ) {
+      throw new Error(
+        'The longest side must be less than the sum of the other sides',
+      );
+    }
+  }
+
+  getArea(): number {
```

Comment:
[CODE STYLE] - `getArea` function doesn't need to accept any parameters. All necessary values are already available as object fields. You should remove the destructuring in line 35, as it is unnecessary.